### PR TITLE
Core - Sign alias transaction

### LINKED ISSUE #359: Core - add Alias transaction tests
## Description

* Core methods should be able to create and sign Alias transactions
* add Alias transaction unit tests

### PR DESCRIPTION
**Pull request by matjazv**
# Description of change

- Fix #359
- Fix Alias Output Clone function 
- Some renaming in `unlock_block.c`

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

New unit tests are in `test_transaction_alias.c`

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/crypto/iota_crypto.h [hunk 1]
```diff
@@ -13,12 +13,12 @@
...
-#define CRYPTO_SHA512_HASH_BYTES 64   // crypto_auth_hmacsha512_BYTES
-#define CRYPTO_SHA256_KEY_BYTES 32    // crypto_auth_hmacsha256_KEYBYTES
-#define CRYPTO_SHA256_HASH_BYTES 32   // crypto_auth_hmacsha256_BYTES
-#define CRYPTO_BLAKE2B_HASH_BYTES 32  // crypto_generichash_blake2b_BYTES
-
+#define CRYPTO_SHA512_KEY_BYTES 32        // crypto_auth_hmacsha512_KEYBYTES
+#define CRYPTO_SHA512_HASH_BYTES 64       // crypto_auth_hmacsha512_BYTES
+#define CRYPTO_SHA256_KEY_BYTES 32        // crypto_auth_hmacsha256_KEYBYTES
+#define CRYPTO_SHA256_HASH_BYTES 32       // crypto_auth_hmacsha256_BYTES
+#define CRYPTO_BLAKE2B_HASH_BYTES 32      // crypto_generichash_blake2b_BYTES
```
Comment (oopsmonk) [inline src/crypto/iota_crypto.h]: for consistancy
```suggestion
#define CRYPTO_BLAKE2B_256_HASH_BYTES 32      // crypto_generichash_blake2b_BYTES
```

**Review by oopsmonk — CHANGES_REQUESTED** [review]

File: src/core/models/signing.h [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ */
+void signing_free(signing_data_list_t* signing_data_list);
+
+/**
+ * @brief Find a signing data by a given index
+ *
+ * @param[in] unlock_address Address Unlock Condition Address - ED25519/NFT/Alias
+ * @param[in] hash Optional, a NFT/Alias ID in the utxo_output
+ * @param[in] hash_len A length of hash array, 0 if hash is NULL
+ * @param[in] keypair TOptional, ed25519 keypair of this input (this is for ed25519 address)
```
Comment (rebinmannady93) [inline src/core/models/signing.h]:
```suggestion
 * @param[in] keypair Optional, ed25519 keypair of this input (this is for ed25519 address)
```

**Review by rebinmannady93 — APPROVED** [review]

File: src/core/models/signing.h (hunk 2 above)
Comment (matjazv) [reply src/core/models/signing.h]: Done. Thanks.

File: src/crypto/iota_crypto.h (hunk 1 above)
Comment (matjazv) [reply src/crypto/iota_crypto.h]: Done. Thanks.

**Review by oopsmonk — APPROVED** [review]

**Pull request merged**